services/provider/server/cephrbdmirror.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const rBDMirrorName = "rbd-mirror"
+
+type cephRBDMirrorManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephRBDMirrorManager(cl client.Client, namespace string) (*cephRBDMirrorManager, error) {
+	return &cephRBDMirrorManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephRBDMirrorManager) Create(ctx context.Context) error {
+
+	cephRBDMirrorObj := &rookCephv1.CephRBDMirror{}
+	err := c.client.Get(ctx, types.NamespacedName{Name: rBDMirrorName, Namespace: c.namespace}, cephRBDMirrorObj)
+	if err == nil {
+		return nil
+	}
+
+	if err != nil && !kerrors.IsNotFound(err) {
+		return err
+	}
+
+	cephRBDMirrorObj.Name = rBDMirrorName
+	cephRBDMirrorObj.Namespace = c.namespace
+	cephRBDMirrorObj.Spec = rookCephv1.RBDMirroringSpec{Count: 1}
+
+	err = c.client.Create(ctx, cephRBDMirrorObj)
+	if err != nil {
+		return err
+	}
```

Comment:
maybe always try to create and fail if error is other than `exists`? or do `if !kerrors.IsNotFound(err) { create }`